.github/workflows/prbot.yml

```diff
@@ -1,18 +1,157 @@
 name: PR Robot
-on:
-- pull_request
-
+on: [pull_request]
 jobs:
-  pr-robot:
+  build:
+    name: "Build"
     runs-on: ubuntu-latest
     env:
       GITHUB_CREDENTIALS: ${{ secrets.GITHUB_TOKEN }}
     steps:
-    - uses: actions/checkout@v2
-      with:
+      - uses: actions/checkout@v2
+        with:
           fetch-depth: 0
-    - uses: actions/setup-node@v1
-      with:
-          node-version: '14'
-    - run: npm i
-    - run: npm run pr:report
+      - uses: actions/setup-node@v2
+        with:
+          node-version-file: '.nvmrc'
+      - uses: actions/cache@v2
```

Comment:
The cache action also executes a post run script to upload to the cache. This significantly reduces NPM install times, cutting them in about half. The cache action works across both jobs and runs, meaning that it should also reduce installation time across different PRs (if I understood correctly).